magma/clock.py

```diff
@@ -95,6 +95,9 @@ def driven_implicitly_by_when(self):
     def wire(self, o, debug_info=None, driven_implicitly_by_when=False):
         if get_debug_mode() and debug_info is None:
             debug_info = get_callee_frame_info()
+        if self._driven_implicitly_by_when and not driven_implicitly_by_when:
+            # User is overriding the implicit when logic
+            self.unwire(debug_info)
```

Comment:
```suggestion
            self.unwire(debug_info)  # user is overriding the implicit when logic
```

(btw "overriding the implicit when logic" is vague and unclear what is actually happening, a more descriptive comment might be more useful).